x-pack/plugins/fleet/server/index.ts

```diff
@@ -17,6 +17,17 @@ import {
 
 import { FleetPlugin } from './plugin';
 
+// Regex to validate if a package name matches <name>:<semver>
+// Valid entries include: abc:1.0.2, def:2.10.3-beta, jkl:3.3.3/subdirectory
+const packageNameWithSemverRegex = () =>
```

Comment:
I would separate the validation of package name and version. The package name validation can be found here: https://github.com/elastic/package-spec/blob/master/versions/1/manifest.spec.yml#L111 The version part here: https://github.com/elastic/package-spec/blob/master/versions/1/manifest.spec.yml#L124

I don't think `jkl:3.3.3/subdirectory` is a valid package?

Reply:
I was basing this on the inclusion of `nginx:1.7.2/logs` in the example config, but I don't think we should go ahead with allowing that convention.